libs/shinkai-message-ts/src/http-client.ts

```diff
@@ -1,5 +1,48 @@
-import axios from 'axios';
+import { invoke } from '@tauri-apps/api/core';
 
-export const httpClient = axios.create({
-  timeout: 2 * 60 * 1000, // 2 minutes
-});
+interface AxiosLikeResponse<T = any> {
+  data: T;
+  status: number;
+  statusText: string;
+  headers: Record<string, string>;
+  config: any;
+  request?: any;
+}
+
+interface AxiosLikeRequestConfig {
+  url?: string;
+  method?: string;
+  headers?: Record<string, string>;
+  data?: any;
+  params?: Record<string, any>;
+  timeout?: number;
+  responseType?: string;
+}
+
+export const httpClient = {
+  async request<T = any>(config: AxiosLikeRequestConfig): Promise<AxiosLikeResponse<T>> {
+    try {
+      const command = config.responseType === 'json' ? 'http_request_json' : 'http_request_string';
```

Comment:
FYI `axios({ url: 'http://potato', method: 'post', data: { test: 1 } })` is json request by default, without setting the responseType.